code/modules/projectiles/updated_projectiles/gun_system.dm

```diff
@@ -466,10 +475,26 @@ load_into_chamber(), reload_into_chamber(), and clear_jam() do all of the heavy
 If you need to change up how a gun fires, just change these procs for that subtype
 and you're good to go.
 */
-/obj/item/weapon/gun/proc/load_into_chamber(mob/user)
+/obj/item/weapon/gun/proc/load_into_chamber(mob/user, overcharge_check = FALSE)
 	//The workhorse of the bullet procs.
  	//If we have a round chambered and no active attachable, we're good to go.
 	if(in_chamber && !active_attachable)
+		if(overcharge_check) //Check to see if we have the proper ammo in chamber to match the overcharge fire mode
```

Comment:
This is a hack to implement energy weapon code on a clearly ballistics system. Use an energy-based system, it will be cleaner and simpler.

Reply:
Let me know which functions/function changes you want me to port over to energy.dm; I think some like the cocking ones and ammo counter feedback are fine.